ch12.md

```diff
@@ -0,0 +1,256 @@
+# Chapter 11: Traversing the Stone
+
+So far, in our cirque du conteneur, you've seen us tame the ferocious [functor](ch8.md#my-first-functor), bending it to our will to perform any operation that strikes our fancy. You've been dazzled by the juggling of many dangerous effects at once using function [application](ch10.md) to collect the results. Sat there in amazement as containers vanished in thin air by [joining](ch9.md) them together. At the side effect sideshow, we've seen them [composed](ch8.md#a-spot-of-theory) into one. And most recently, we've ventured beyond what's natural and [transformed](ch11.md) one type into another before your very eyes.
+
+And now for our next trick, we'll look at traversals. We'll watch types soar over one another as if they were trapeze artists holding our value in tact. We'll reorder effects like the trolleys in a tilt-a-whirl. When our containers get intertwined like the limbs of a contortionist, we can use this interface to straighten things out. We'll witness different effects with different orderings. Fetch me my pantaloons and slide whistle, let's get started.
+
+## Types n' types
+
+Let's get weird:
+
+```js
+// readFile :: FileName -> Task Error String
+
+// firstWords :: String -> String
+const firstWords = compose(join(' '), take(3), split(' '))
+
+// tldr :: FileName -> Task Error String
+const tldr = compose(map(firstWords), readFile)
+
+map(tldr, files)
+// [Task('hail the monarchy'), Task('smash the patriarchy')]
+```
+
+Here we read a bunch of files and end up with a useless array of tasks. How might we fork each one of these? It would be most agreeable if we could switch the types around to have `Task Error [String]` instead of `[Task Error String]`. That way, we'd have one future value holding all the results, which is much more amendable to our async needs than several future values arriving at their leisure.
+
+Here's one last example of a sticky situation:
+
+```js
+// getAttribute :: String -> Node -> Maybe String
+// $ :: Selector -> IO Node
+
+// getControlNode :: IO (Maybe (IO Node))
+const getControlNode = compose(map(map($)), map(getAttribute('aria-controls')), $)
```

Comment:
Should this show some arbitrary the result? or is it more let the reader follow it as an exercise?

Reply:
I debated calling it with some value to demonstrate, but felt it blurred the focus since I wanted to highlight the type signature. What do you think? It might be too abstract without.